### ISSUE TITLE
[Bug]: Continuous access to abnormal ObServer which is restarting

### ISSUE THREAD

**Issue by MachaelLee** [oceanbase/obkv-table-client-rs#19]
### Check Before Asking

- [X] Please check the issue list and confirm this bug is encountered for the first time.
- [X] Please try full text in English and attach precise description.

### Environment

```markdown
* version:
name = "obkv-table-client-rs"
version = "0.1.0"
source = "git+https://github.com/oceanbase/obkv-table-client-rs.git?rev=211e5718630577a7f8c1a2d74055bad4d31dea57#211e5718630577a7f8c1a2d74055bad4d31dea57"
```

### Fast Reproduce Steps

1. restart one ObServer
2. restart client

### Actual Behavior

* stacktrace
```
2023-03-08 21:59:43.979 ERRO [server/src/grpc/storage_service/mod.rs:237] Failed to handle request, mod:write, handler:handle_write, err:Rpc error, code:500, message:failed to execute interpreter, cause:Failed to execute insert, err:Failed to write table, err:Failed to write table, table:SPM_1113008887_INFLUENCE_DEFAULT, err:Failed to write table, source:Channel error, err:Background flush failed, cannot write more data, err:Failed to store version edit, err:Failed to write update to wal, err:Failed to write log entries, err:Failed to write table unit, namespace:manifest, wal location:WalLocation { versioned_region_id: VersionedRegionId { version: 1, id: 112 }, table_id: 8977 }, err:Failed to write log to table, region_id:112, err:Failed to write to table, table:wal_manifest_permanent_000112, err:Common error, code:ObException(OB_NOT_MASTER), err:rcode:OB_NOT_MASTER, message:, addr:33.144.36.65:2882, trace_id:Y220D7D39-0000000000000001, server_trace_id:Y220D7D39-0000000000000001.
Backtrace:
0 <snafu::backtrace_shim::Backtrace as snafu::GenerateBacktrace>::generate::he03e4c3ab1c80600
  /root/.cargo/registry/src/github.com-1ecc6299db9ec823/snafu-0.6.10/src/backtrace_shim.rs:15
  <table_kv::obkv::WriteTable<__T0> as snafu::IntoError<table_kv::obkv::Error>>::into_error::h4bfe386477d0d25d
  /xunming/ceresdb/components/table_kv/src/obkv.rs:23
  <core::result::Result<T,E> as snafu::ResultExt<T,E>>::context::{{closure}}::hf9392c06be72b7f7
  /root/.cargo/registry/src/github.com-1ecc6299db9ec823/snafu-0.6.10/src/lib.rs:318
  core::result::Result<T,E>::map_err::h0a680db20734d2fb
  /rustc/11d96b59307b1702fffe871bfc2d0145d070881e/library/core/src/result.rs:860
  <core::result::Result<T,E> as snafu::ResultExt<T,E>>::context::he43d36ef3245f2a9
  /root/.cargo/registry/src/github.com-1ecc6299db9ec823/snafu-0.6.10/src/lib.rs:318
  <table_kv::obkv::ObkvImpl as table_kv::TableKv>::write::hd201be4265267d92
  /xunming/ceresdb/components/table_kv/src/obkv.rs:458
1 wal::table_kv_impl::table_unit::TableUnitWriter::write_log::{{closure}}::{{closure}}::ha7d265dcbe3da346
  /xunming/ceresdb/wal/src/table_kv_impl/table_unit.rs:944
  <tokio::runtime::blocking::task::BlockingTask<T> as core::future::future::Future>::poll::ha71b92e94a88d4b2
  /root/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.25.0/src/runtime/blocking/task.rs:42
  tokio::runtime::task::core::Core<T,S>::poll::{{closure}}::h4a34186a5b0bfcc5
  /root/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.25.0/src/runtime/task/core.rs:223
  tokio::loom::std::unsafe_cell::UnsafeCell<T>::with_mut::hdea7b572d86fc8e1
  /root/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.25.0/src/loom/std/unsafe_cell.rs:14
  tokio::runtime::task::core::Core<T,S>::poll::h640afc6a912254f4
  /root/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.25.0/src/runtime/task/core.rs:212
  tokio::runtime::task::harness::poll_future::{{closure}}::hcbce8b5c4b437606
  /root/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.25.0/src/runtime/task/harness.rs:476
  <core::panic::unwind_safe::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once::h5c1bf2ced7c4e5e4
  /rustc/11d96b59307b1702fffe871bfc2d0145d070881e/library/core/src/panic/unwind_safe.rs:271
  std::panicking::try::do_call::hed3b2255ca0ddaa5
  /rustc/11d96b59307b1702fffe871bfc2d0145d070881e/library/std/src/panicking.rs:483
  std::panicking::try::hfd3bbb404eaf6d7f
  /rustc/11d96b59307b1702fffe871bfc2d0145d070881e/library/std/src/panicking.rs:447
  std::panic::catch_unwind::hc7b46f890d6135d1
  /rustc/11d96b59307b1702fffe871bfc2d0145d070881e/library/std/src/panic.rs:140
  tokio::runtime::task::harness::poll_future::h81880f781222b50f
  /root/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.25.0/src/runtime/task/harness.rs:464
  tokio::runtime::task::harness::Harness<T,S>::poll_inner::hf12ee18d37ad3280
  /root/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.25.0/src/runtime/task/harness.rs:198
  tokio::runtime::task::harness::Harness<T,S>::poll::hf678f6d25da27707
  /root/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.25.0/src/runtime/task/harness.rs:152
  tokio::runtime::task::raw::poll::hfca250bd203e01f9
  /root/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.25.0/src/runtime/task/raw.rs:255
2 tokio::runtime::task::raw::RawTask::poll::hee1eeac52e93dc2f
  /root/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.25.0/src/runtime/task/raw.rs:200
  tokio::runtime::task::UnownedTask<S>::run::h58fd606513e73ddb
  /root/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.25.0/src/runtime/task/mod.rs:431
  tokio::runtime::blocking::pool::Task::run::h9890a5808360982f
  /root/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.25.0/src/runtime/blocking/pool.rs:159
  tokio::runtime::blocking::pool::Inner::run::h9c0e0c521fb2e6f9
  /root/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.25.0/src/runtime/blocking/pool.rs:511
  tokio::runtime::blocking::pool::Spawner::spawn_thread::{{closure}}::h18b2e71018da51d1
  /root/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.25.0/src/runtime/blocking/pool.rs:469
  std::sys_common::backtrace::__rust_begin_short_backtrace::h8516ed23210cf795
  /rustc/11d96b59307b1702fffe871bfc2d0145d070881e/library/std/src/sys_common/backtrace.rs:121
3 std::thread::Builder::spawn_unchecked_::{{closure}}::{{closure}}::h0601890e5bbaea04
  /rustc/11d96b59307b1702fffe871bfc2d0145d070881e/library/std/src/thread/mod.rs:558
  <core::panic::unwind_safe::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once::h7b1241f8fb072088
  /rustc/11d96b59307b1702fffe871bfc2d0145d070881e/library/core/src/panic/unwind_safe.rs:271
  std::panicking::try::do_call::he655ac3b8c52908c
  /rustc/11d96b59307b1702fffe871bfc2d0145d070881e/library/std/src/panicking.rs:483
  std::panicking::try::h9a85ae35c2c9342a
  /rustc/11d96b59307b1702fffe871bfc2d0145d070881e/library/std/src/panicking.rs:447
  std::panic::catch_unwind::h56fd0ce15d2a37b2
  /rustc/11d96b59307b1702fffe871bfc2d0145d070881e/library/std/src/panic.rs:140
  std::thread::Builder::spawn_unchecked_::{{closure}}::h86790adc1395a6d6
  /rustc/11d96b59307b1702fffe871bfc2d0145d070881e/library/std/src/thread/mod.rs:557
  core::ops::function::FnOnce::call_once{{vtable.shim}}::h5edf909cb71465eb
  /rustc/11d96b59307b1702fffe871bfc2d0145d070881e/library/core/src/ops/function.rs:250
4 <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once::hc8beb91c5e39b692
  /rustc/11d96b59307b1702fffe871bfc2d0145d070881e/library/alloc/src/boxed.rs:1988
  <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once::h20e58dce1054acc4
  /rustc/11d96b59307b1702fffe871bfc2d0145d070881e/library/alloc/src/boxed.rs:1988
  std::sys::unix::thread::Thread::new::thread_start::h848946a57aa81736
  /rustc/11d96b59307b1702fffe871bfc2d0145d070881e/library/std/src/sys/unix/thread.rs:108
5 start_thread
6 __clone
.
Backtrace:
 0 <snafu::backtrace_shim::Backtrace as snafu::GenerateBacktrace>::generate::he03e4c3ab1c80600
   /root/.cargo/registry/src/github.com-1ecc6299db9ec823/snafu-0.6.10/src/backtrace_shim.rs:15
   <wal::manager::error::Write as snafu::IntoError<wal::manager::error::Error>>::into_error::hdda3710b1a82a620
   /xunming/ceresdb/wal/src/manager.rs:30
 1 <core::result::Result<T,E> as snafu::ResultExt<T,E>>::context::{{closure}}::h0a848c8c496e20d7
   /root/.cargo/registry/src/github.com-1ecc6299db9ec823/snafu-0.6.10/src/lib.rs:318
   core::result::Result<T,E>::map_err::hd6b4b64c6016aa5e
   /rustc/11d96b59307b1702fffe871bfc2d0145d070881e/library/core/src/result.rs:860
   <core::result::Result<T,E> as snafu::ResultExt<T,E>>::context::h1f1f32c573b1b9df
   /root/.cargo/registry/src/github.com-1ecc6299db9ec823/snafu-0.6.10/src/lib.rs:318
   <wal::table_kv_impl::wal::WalNamespaceImpl<T> as wal::manager::WalManager>::write::{{closure}}::h21a4555b31bf8398
   /xunming/ceresdb/wal/src/table_kv_impl/wal.rs:155
 2 <core::pin::Pin<P> as core::future::future::Future>::poll::hbb0f639ad60877c8
   /rustc/11d96b59307b1702fffe871bfc2d0145d070881e/library/core/src/future/future.rs:125
   <analytic_engine::manifest::details::WalBasedLogStore as analytic_engine::manifest::details::MetaUpdateLogStore>::append::{{closure}}::h9df6173a6830e87c
   /xunming/ceresdb/analytic_engine/src/manifest/details.rs:503
 3 <core::pin::Pin<P> as core::future::future::Future>::poll::hfd522eb5853f1b83
   /rustc/11d96b59307b1702fffe871bfc2d0145d070881e/library/core/src/future/future.rs:125
   analytic_engine::manifest::details::ManifestImpl::store_update_to_wal::{{closure}}::hcf471e5adca29e87
   /xunming/ceresdb/analytic_engine/src/manifest/details.rs:230
   <analytic_engine::manifest::details::ManifestImpl as analytic_engine::manifest::Manifest>::store_update::{{closure}}::h96a56ce12c3db98a
   /xunming/ceresdb/analytic_engine/src/manifest/details.rs:302
 4 <core::pin::Pin<P> as core::future::future::Future>::poll::hfa9005ee6ffdbd04
   /rustc/11d96b59307b1702fffe871bfc2d0145d070881e/library/core/src/future/future.rs:125
   analytic_engine::instance::flush_compaction::<impl analytic_engine::instance::Instance>::dump_memtables::{{closure}}::h3b62ceebb586c5cd
   /xunming/ceresdb/analytic_engine/src/instance/flush_compaction.rs:564
   analytic_engine::instance::flush_compaction::<impl analytic_engine::instance::Instance>::flush_memtables::{{closure}}::h97f0628210479467
   /xunming/ceresdb/analytic_engine/src/instance/flush_compaction.rs:426
   analytic_engine::instance::flush_compaction::<impl analytic_engine::instance::Instance>::schedule_table_flush::{{closure}}::{{closure}}::h6b2966852d6514f7
   /xunming/ceresdb/analytic_engine/src/instance/flush_compaction.rs:351
 5 analytic_engine::instance::write_worker::WorkerLocal::flush_sequentially::{{closure}}::{{closure}}::h0ae5bb52ad3c6ad9
   /xunming/ceresdb/analytic_engine/src/instance/write_worker.rs:261
 6 tokio::runtime::task::core::Core<T,S>::poll::{{closure}}::heb6a096dd70c72c7
   /root/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.25.0/src/runtime/task/core.rs:223
   tokio::loom::std::unsafe_cell::UnsafeCell<T>::with_mut::hd9d26565fb8b177a
   /root/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.25.0/src/loom/std/unsafe_cell.rs:14
   tokio::runtime::task::core::Core<T,S>::poll::h8ed1db6786089356
   /root/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.25.0/src/runtime/task/core.rs:212
   tokio::runtime::task::harness::poll_future::{{closure}}::hb7b8af20497f9bf8
   /root/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.25.0/src/runtime/task/harness.rs:476
   <core::panic::unwind_safe::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once::hc3657e4730af664f
   /rustc/11d96b59307b1702fffe871bfc2d0145d070881e/library/core/src/panic/unwind_safe.rs:271
   std::panicking::try::do_call::hff3793ccfad12d19
   /rustc/11d96b59307b1702fffe871bfc2d0145d070881e/library/std/src/panicking.rs:483
   std::panicking::try::h8f8037b0d98c30a9
   /rustc/11d96b59307b1702fffe871bfc2d0145d070881e/library/std/src/panicking.rs:447
   std::panic::catch_unwind::h22c8eb1e308bf65c
   /rustc/11d96b59307b1702fffe871bfc2d0145d070881e/library/std/src/panic.rs:140
   tokio::runtime::task::harness::poll_future::h2ca8fe85f3fcdc98
   /root/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.25.0/src/runtime/task/harness.rs:464
   tokio::runtime::task::harness::Harness<T,S>::poll_inner::h09184acd970dd1dc
   /root/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.25.0/src/runtime/task/harness.rs:198
   tokio::runtime::task::harness::Harness<T,S>::poll::h5b025d4888aca5f8
```

### Expected Behavior

Access to other OB servers

**Comment by ShiKaiWi** [comment]
After digging into the codebase, it is found that the initializing server is marked as active on the inner route table. And the connection pool manager will panic if failed to connect to some server when start.

Maybe we should ignore such connecting failure if any one server is accessible.

**Comment by MachaelLee** [comment]
fixed in #21
